todo/src/components/Form.js

```diff
@@ -0,0 +1,53 @@
+import React from "react"
+import {connect} from "react-redux";
+import {add, clear} from '../actions/actions';
+
+
+class Form extends React.Component {
+   constructor(props){
+      super(props)
+      this.state = {
+         input: "",
+      }
+   }
+
+   inputHandler = (e) => {
+      this.setState({
+         input: e.target.value,
+      })
+   }
+
+   addHandler = () => {
+      this.props.add(this.state.input)
+      this.setState({
+         input: "",
+      })
+   }
+
+   clearCompleted = () => {
+      this.props.clear()
+   }
+    render(){
+      return(
+         <div>
+            <form onSubmit={(e) => e.preventDefault()}>
```

Comment:
This should probably be in a submit handler and then preventDefault in that method. Then you wouldn't need to check for the onKeyDown event